File: test/EntityFramework/VBTests/_VbMonsterModel/ExternalTypes/LoginMm.vb
'---------------------------------------------------------------------
' <copyright file="LoginMm.vb" company="Microsoft">
'      Copyright (c) Microsoft Corporation.  All rights reserved.
' </copyright>
'
' @owner       avickers
' @backupOwner bricelam
'---------------------------------------------------------------------

Imports System
Imports System.Collections.Generic

Namespace Another.Place
    Partial Public Class LoginMm
        Public Property Username As String
        Public Property CustomerId As Integer

        Public Overridable Property Customer As CustomerMm
        Public Overridable Property LastLogin As LastLoginMm
        Public Overridable Property SentMessages As ICollection(Of MessageMm) = New HashSet(Of MessageMm)
        Public Overridable Property ReceivedMessages As ICollection(Of MessageMm) = New HashSet(Of MessageMm)
        Public Overridable Property Orders As ICollection(Of OrderMm) = New HashSet(Of OrderMm)

    End Class
End Namespace

File: test/EntityFramework/FunctionalTests/TestModels/TemplateModels/CsMonsterModel/ExternalTypes/LoginMm.cs
// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved. See License.txt in the project root for license information.

namespace Another.Place
{
    using System.Collections.Generic;
    using FunctionalTests.ProductivityApi.TemplateModels.CsMonsterModel;

    public class LoginMm
    {
        public LoginMm()
        {
            SentMessages = new HashSet<MessageMm>();
            ReceivedMessages = new HashSet<MessageMm>();
            Orders = new HashSet<OrderMm>();
        }

        public string Username { get; set; }
        public int CustomerId { get; set; }

        public virtual CustomerMm Customer { get; set; }
        public virtual LastLoginMm LastLogin { get; set; }
        public virtual ICollection<MessageMm> SentMessages { get; set; }
        public virtual ICollection<MessageMm> ReceivedMessages { get; set; }
        public virtual ICollection<OrderMm> Orders { get; set; }
    }
}
